py-polars/polars/dataframe/frame.py

```diff
@@ -3440,7 +3449,7 @@ def write_ipc_stream(
         if compression is None:
             compression = "uncompressed"
 
-        if future:
+        if future is True:
```

Comment:
The warning must be issued also if `future` is an integer.

Reply:
I don't think so. A specific version is not unstable. e.g. `future=1` is forward compatible giving all breaking changes will introduce a new version (`future=2`).